aurora/src/main/java/src/handlers/SubscribeHandler.java

```diff
@@ -27,43 +29,28 @@ public SubscribeHandler(ChannelManager channelManager) {
 
     public void handleSubscribe(Aurora.AuroraRequest request, StreamObserver<Aurora.AuroraResponse> responseObserver) {
         LOGGER.info("Handling subscribe from client {}", request.getClientId());
-        Optional<ManagedChannel> channelByKey = channels.getChannelByKey(request.getTopic().split("/")[0]);
-        if (channelByKey.isPresent()) {
-            AuroraServiceGrpc.AuroraServiceStub stub = generateAuroraStub(channelByKey.get());
-
-            startStream(request, stub, responseObserver);
-        } else {
+        List<ManagedChannel> channelsWithPrefix = this.channels.getAllChannelsContainingPrefix(request.getTopic().split("/")[0]);
+        if (channelsWithPrefix.isEmpty()) {
             LOGGER.warn("Unsupported message have reached aurora.");
             responseObserver.onError(Status.INVALID_ARGUMENT
                     .withDescription(request.getTopic().split("/")[0] + " channel not available at the moment.")
                     .asException());
+            return;
+        }
+        CountDownLatch latch = new CountDownLatch(channelsWithPrefix.size());
+
+        channelsWithPrefix.forEach(channel -> this.generateAuroraStub(channel)
+                .subscribe(request, new AuroraObserver(request, responseObserver, latch)));
+
+        try {
+            latch.await();
```

Comment:
I am wondering if it is a good idea to have many waiting threads. What about using atomic integer that will be decreased and if its value is 0 then onCompleted will be executed

Reply:
and do checks if it's 0 in endless loop ?

Reply:
fixed